File: 0020. Valid Parentheses/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = list()
        
        for ch in s:
            if ch is '(':
                stack.append(')')
            elif ch is '[':
                stack.append(']')
            elif ch is '{':
                stack.append('}')
            else:
                if ch is not ')' and ch is not ']' and ch is not '}':
                    continue
                elif not stack or stack[-1] != ch:
                    return False
                else:
                    stack.pop()
        
        return not stack

File: 0020. Valid Parentheses/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        
        for (auto ch : s) {
            if (ch == '(') st.push(')');
            else if (ch == '[') st.push(']');
            else if (ch == '{') st.push('}');
            else {
                if (ch != ')' && ch != ']' && ch != '}') continue;
                else if (st.empty() || st.top() != ch) return false;
                else st.pop();
            }
        }
        
        return st.empty();
    }
};